src/Google.Bigquery.V2/BigqueryClient.DatasetCrud.cs

```diff
@@ -0,0 +1,160 @@
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Apis.Bigquery.v2.Data;
+using System;
+using System.Collections.Generic;
+
+namespace Google.Bigquery.V2
+{
+    public partial class BigqueryClient
+    {
+        /// <summary>
+        /// Retrieves a dataset within this client's project given the dataset ID.
+        /// This method just creates a <see cref="DatasetReference"/> and delegates to <see cref="GetDataset(DatasetReference)"/>.
+        /// </summary>
+        /// <param name="datasetId">The dataset ID. Must not be null.</param>
+        /// <returns>The requested dataset.</returns>
+        public virtual BigqueryDataset GetDataset(string datasetId) =>
+            GetDataset(GetDatasetReference(datasetId));
+
+        /// <summary>
+        /// Retrieves a dataset given a project ID and dataset ID.
+        /// This method just creates a <see cref="DatasetReference"/> and delegates to <see cref="GetDataset(DatasetReference)"/>.
+        /// </summary>
+        /// <param name="projectId">The project ID. Must not be null.</param>
+        /// <param name="datasetId">The dataset ID. Must not be null.</param>
+        /// <returns>The requested dataset.</returns>
+        public virtual BigqueryDataset GetDataset(string projectId, string datasetId) =>
+            GetDataset(GetDatasetReference(projectId, datasetId));
+
+        /// <summary>
+        /// Retrieves a dataset.
+        /// </summary>
+        /// <param name="datasetReference">A fully-qualified identifier for the dataset. Must not be null.</param>
+        /// <returns>The requested dataset.</returns>
+        public virtual BigqueryDataset GetDataset(DatasetReference datasetReference)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Lists the datasets within this client's project.
+        /// This method just delegates to <see cref="ListDatasets(string, ListDatasetsOptions)"/>.
+        /// </summary>
+        /// <param name="options">The options for the operation. May be null, in which case
+        /// defaults will be supplied.</param>
+        /// <returns>The datasets within this project.</returns>
+        public virtual IEnumerable<BigqueryDataset> ListDatasets(ListDatasetsOptions options = null)
+            => ListDatasets(ProjectId, options);
+
+        // TODO: Take a ProjectReference instead? Seems a bit pointless.
```

Comment:
I think (at the moment, at any rate) that ProjectReference should be used all through this in a manner consistent with the other Reference types.
Some users may well wish to use the various Reference classes consistently throughout their code.
And these methods are easily ignorable if someone doesn't want to use them.